Frontend/ui-driver/src/Helpers/Utils.purs

```diff
@@ -239,12 +239,46 @@ capitalizeFirstChar inputStr =
     in DS.joinWith " " output
 
 getDowngradeOptions :: String -> Array String
-getDowngradeOptions variant = case variant of
-                                "TAXI"  -> []
-                                "SUV"   -> ["SEDAN", "HATCHBACK"]
-                                "SEDAN" -> ["TAXI", "HATCHBACK"] 
-                                _       -> ["TAXI"]
-
+getDowngradeOptions variant = case (getMerchant FunctionCall) of
+                                YATRISATHI -> case variant of
+                                                "TAXI"  -> []
+                                                "SUV"   -> ["SEDAN", "HATCHBACK"]
+                                                "SEDAN" -> ["TAXI", "HATCHBACK"] 
+                                                _       -> ["TAXI"]
+                                _ -> case variant of
+                                        "SUV"   -> ["SEDAN", "HATCHBACK"]
+                                        "SEDAN" -> ["HATCHBACK"]
+                                        _       -> []
+
+
+getDowngradeOptionsText :: String -> String
+getDowngradeOptionsText vehicleType = do
+  let
+    downgradeFrom = case (getMerchant FunctionCall) of
+                    YATRISATHI -> case vehicleType of
+                                    "SUV" -> getString AC_SUV
+                                    _     -> getString AC_CAB
+                    _ -> getVehicleType vehicleType
+
+    downgradeOptions = getUIDowngradeOptions vehicleType
+    downgradeTo = case downgradeOptions DA.!! 0 of
+                        Just option -> getVariantRideType option
+                        Nothing      -> ""
+  getString DOWNGRADING_VEHICLE_WILL_ALLOW_YOU_TO_TAKE_BOTH_1 <> if ((getMerchant FunctionCall) == YATRI && vehicleType == "SUV") then downgradeFrom <> ", " <> getString SEDAN <> getString AND <> getString HATCHBACK <> getString DOWNGRADING_VEHICLE_WILL_ALLOW_YOU_TO_TAKE_BOTH_3
+                                                                    else downgradeFrom <> getString AND <> downgradeTo <> getString DOWNGRADING_VEHICLE_WILL_ALLOW_YOU_TO_TAKE_BOTH_3
```

Comment:
why we need to add check here based on variant and merchant, instead we can just call the `getUIDowngradeOptions ` function to get the array of variant we can downgrade to and handle the `AND` and `,` based on the array size. This will make it configurable for both the merchants